### PR TITLE
#6: add indexing to BodyOfTheLastResponse.

### PR DESCRIPTION
**Pull request by perrygoy**
This PR adds `__getitem__` handling for `BodyOfTheLastResponse`, which does two things:
1. Adds the requested index to the list of `body_parts` (sorry i couldn't help myself).
2. Returns `self`.

This means further indexing will be kept track of in the `body_parts` list. As the question is answered, `BodyOfTheLastResponse` checks if it has any `body_parts`, and will travel down the list for the user while answering the Question.

One thing for discussion—should a `KeyError` or `IndexError` just be let to bubble up, or should we catch that and tell the user what went wrong?

### CODE REVIEW

File: screenpy_requests/actions/__init__.py [hunk 1]
```diff
@@ -23,11 +23,11 @@ def generate_send_method_class(method: str) -> Type[APIMethodAction]:
     """
 
     class SendMETHODRequest:
-        "Will be programmatically replaced."
+        """Will be programmatically replaced."""
```
Comment (perrygoy) [inline screenpy_requests/actions/__init__.py:26]: Oh yeah also i had this fix here for months. Haha.

File: screenpy_requests/questions/body_of_the_last_response.py [hunk 2]
```diff
@@ -15,20 +17,49 @@
...
+
+        # text response, sliced
+        the_actor.should(
+            See.the(BodyOfTheLastResponse()[6:12], ReadsExactly("Python"))
+        )
     """
 
+    body_parts: list[str | int]
+
+    def __getitem__(self, key: str | int) -> BodyOfTheLastResponse:
```
Comment (MarcelWilson) [inline screenpy_requests/questions/body_of_the_last_response.py]: Add `slice` to `key` and `body_parts` annotations
Mypy will bitch about the following:
`foo = BodyOfTheLastResponse()["slideshow"]["slides"][1]["items"][0:2]`

File: screenpy_requests/actions/__init__.py (hunk 1 above)
Comment (MarcelWilson) [reply screenpy_requests/actions/__init__.py:26]: I figure when we add ruff to the project, all these would be updated anyway.

File: tests/test_questions.py [hunk 3]
```diff
@@ -53,6 +52,32 @@ def test_ask_for_body_of_the_last_response(self, APITester):
 
         assert BodyOfTheLastResponse().answered_by(APITester) == test_json
 
+    def test_stores_index_path(self):
+        botlr = BodyOfTheLastResponse()["shuffled"]["off"][10]["mortal"]["coils"]
```
Comment (MarcelWilson) [inline tests/test_questions.py:56]: I would add a slice here.

**Comment by MarcelWilson** [comment]
Looks good!  Only have one last suggestion: add a slice example to  `test_stores_index_path`

**Review by bandophahita — APPROVED** [review]
Very clever sir.

**Pull request merged**